### DISCUSSION TITLE
Power in

### DISCUSSION THREAD

**Discussion by alpapak** [PiSupply/PiJuice#707] [Q&A] [answered]
Is it possible to have power in on Pi and at the same time a second adaptor to power in the pijuice?
Pijuice will get from two source power.

What will happen if i connect both source at the same time?

Finally, can i charge and at the same time have my pi on from a power bank? My power bank output is 3A.

**Comment by tvoverbeek** [comment, marked as answer]
Yes, both power inputs may be connected. You can specify which of the two inputs has preference in the GUI/.CLI.
3A is sufficient to both power the Pi and charge (assuming no power hungry accessories are connected).

**Reply by alpapak** [reply]
Thank you. I am using CLI, where can i set up this? Can i have Main and secondary power in source? I would like to have Main power In on my Pi and add a power bank as secondary power in connected to my pijuice. In case power lost to continue from my power bank(if it drain the battery i will still have the pijuice battery to safe shutdown) and if power is on again to charge from Pi.

**Reply by lcnittl** [reply]
> I am using CLI, where can i set up this?

This should be `General` > `Inputs precedence`. There choose either `USB_MICRO` to prefer input from the PiJuice HAT's USB port, or `5V_GPIO` to prefer it from the Pi's USB port (from where it reaches the HAT through GPIO).

**Reply by alpapak** [reply]
So, I can connect Pi to power and PiJuice to a power bank. 
My set up will be in CLI > 5V_GPIO. When, the power is off, the PiJuice will continue to work from the power bank until the battery drain. Then, i will have the PiJuice battery until the Pi safe shutdown. Meanwhile, If power is back on, Pi and PiJuice will power from usb-c(5v_GPIO). Correct?

```
Wall adaptor        Pi4              PiJuice               Power Bank    Sonoff       Wall Charger
        X---------->XXXX---5V_GPIO---XXXXXXX<---usb micro---XXXXXXXX<----XXXXX<---------XXXXXXX
```

I will use a sonoff usb 5v wifi, in order to schedule every night or manually, to charge the power bank for 2h. 
[https://sonoff.tech/product/diy-smart-switch/micro/](url)

ps: My power bank will be anker.
[https://us.anker.com/collections/portable-power/products/a1231-1](url)
